### ISSUE TITLE
[New Project]: Train Your Own Brain

### ISSUE THREAD

**Issue by pavanakarthik12** [YadavAkhileshh/OpenPlayground#962]
### Project Name

BrainForge

### Category

experimental

### Project Description

BrainForge is an interactive, browser-based deep learning sandbox built entirely with HTML, CSS, and vanilla JavaScript.
It allows users to visually design, train, and experiment with neural networks, making deep learning concepts tangible and intuitive.

Instead of hiding complexity behind libraries, BrainForge exposes the core mechanics of deep learning—neurons, weights, activations, loss, and backpropagation—through live visual feedback and playful interaction.
The project blends education, experimentation, and fun, helping users feel how neural networks learn rather than just reading about them.

### Key Features

Visual Neural Network Builder

Add and remove layers dynamically

Adjust number of neurons per layer

Drag-and-drop style architecture design

2. Real-Time Training Visualization

Forward propagation animated through the network

Backpropagation shown as error signals flowing backward

Weight updates visualized using color and thickness

3. Activation Function Control

Choose activation functions per layer

ReLU

Sigmoid

Tanh

Immediate visual feedback on how activations affect learning

4. Live Loss & Performance Graphs

Real-time loss curve during training

Clear visualization of:

Convergence

Divergence

Overfitting behavior

5. Interactive Dataset Playground

Built-in datasets (XOR, AND, OR)

Custom user-defined input points

Instantly retrain models on new data

6. Adjustable Training Parameters

Learning rate control

Epoch speed adjustment

Pause, resume, and reset training

### Screenshots

**### UI will be upgraded**
<img width="896" height="849" alt="Image" src="https://github.com/user-attachments/assets/715f290c-1893-4c65-a589-f50c3afbc84e" />

### Checklist

- [x] My project uses vanilla HTML, CSS, and JavaScript
- [x] I have tested my project locally
- [x] My project is responsive
- [x] I will add my project to projects.json (NOT index.html)
